src/main/java/com/spotify/autoscaler/ClusterStats.java

```diff
@@ -279,18 +156,6 @@ void setStats(BigtableCluster cluster, int currentNodes) {
                       : 0;
                 });
       }
-
-      this.registry.register(
-          APP_PREFIX
-              .tagged("what", "cpu-target-ratio")
-              .tagged("project-id", cluster.projectId())
-              .tagged("cluster-id", cluster.clusterId())
-              .tagged("instance-id", cluster.instanceId()),
-          (Gauge<Double>)
-              () -> {
-                final ClusterData data = registeredClusters.get(cluster.clusterName());
-                return data.getCpuUtil() / cluster.cpuTarget();
-              });
     } else {
       clusterData.setCurrentNodeCount(currentNodes);
```

Comment:
this is not related to this pr, but i think there should be a `clusterData.setCluster` here, because the configs might have changed, like the `cpuTarget` for example. then the cpu_target_ratio metric above becomes wrong.

Reply:
I'm finding this aspect hard to follow. I might be misreading the current code, so this might not make sense:

My opinion is that we should confine all shared mutable state down to the one `ConcurrentHashMap<String, ClusterData>` of _immutable_ `ClusterData` objects. The lambdas shouldn't capture anything but the key to this map, so for example not the `cluster.cpuTarget()`. `setStats` shouldn't lookup a `ClusterData` and mutate it; it should replace it with a new `ClusterData`. Once one of these lambdas looks up the `ClusterData` in the map, that should contain all the information that is needed to report a metric.

Reply:
yeah that sounds good too. then the `putIfAbsent` should be replaced with a `put`, and also the `setLoad` method should change to sth similar.

Reply:
I see your (scio) point here, but do you feel that we should add this change in this PR?

Reply:
I think we can simply add `clusterData.setCluster` here, and do the bigger refactor in another pr. wdyt @Bj0rnen ?

Reply:
I added the setCluster, let's discuss this better tomorrow and not merge it today :)